store/sqlstore/channel_store.go

```diff
@@ -521,9 +481,16 @@ func (s SqlChannelStore) createInitialSidebarCategoriesT(transaction *gorp.Trans
 	}
 
 	if !hasCategoryOfType(model.SidebarCategoryFavorites) {
+		favoritesCategoryId := model.NewId()
+
+		// Create the SidebarChannels first since there's more opportunity for something to fail heree
+		if err := s.migrateFavoritesToSidebarT(transaction, userId, teamId, favoritesCategoryId); err != nil {
+			return errors.Wrap(err, "createInitialSidebarCategoriesT: failed to migrate favorites to sidebar")
+		}
+
 		if err := transaction.Insert(&model.SidebarCategory{
 			DisplayName: "Favorites", // This will be retranslated by the client into the user's locale
```

Comment:
Do we have a database-level constraint on `DisplayName`? Worried that a racey double write for the first customer will result in duplicate `Favorites` sidebar categories.

Reply:
We tried adding a unique constraint on the `Type` field, but we weren't able to do it because users can have multiple "custom" categories. I looked into a partial unique constraint, but it's unfortunately not supported on MySQL. `hasCategoryOfType` and running this in a transaction should hopefully mitigate against it at least.

Reply:
I'm fine with accepting the risk, but just want to note that without a unique constraint, a transaction won't guard against this at all.